src/sliding_sync.rs

```diff
@@ -675,6 +749,62 @@ async fn async_worker(
                     // log!("Note: typing notifications recv loop has ended for room {}", room_id);
                 });
             }
+            MatrixRequest::SubscribeToUpdates { room_id, subscribe } => {
+                let mut update_receiver = {
+                    let mut all_room_info = ALL_ROOM_INFO.lock().unwrap();
+                    let Some(room_info) = all_room_info.get_mut(&room_id) else {
+                        log!("BUG: room info not found for subscribe to updates, room {room_id}");
+                        continue;
+                    };
+                    let recv = if subscribe {
+                        if room_info.update_subscriber_initialised {
+                            warning!("Note: room {room_id} is already subscribed to updates.");
+                            continue
+                        }
+                        let Some(room) = CLIENT.get().and_then(|c| c.get_room(&room_id)) else {
+                            error!("BUG: client/room not found when subscribing to updates, room: {room_id}");
+                            continue;
+                        };
+                        room_info.update_subscriber_initialised = true;
+                        room.subscribe_to_updates()
+                    } else {
+                        continue;
+                    };
+                    recv
+                };
+
+                let _to_updates_task = Handle::current().spawn(async move {
+                    while let Ok(room_update) = update_receiver.recv().await {
+                        if let RoomUpdate::Joined { room, updates: _ } = room_update {
+                            // Using Serde_json to obtain the latest_active event_id  
+                            // Room's read receipt's latest_active event is not public, but the ReadReceipt struct 
+                            // is serializable 
+                            if let Some(latest_active) = serde_json::to_value(room.read_receipts())
+                                .unwrap_or_default()
+                                .get("latest_active")
+                                .unwrap_or(&serde_json::Value::Null)
+                                .get("event_id") {
```

Comment:
this is unreadable without proper formatting
```suggestion
                            if let Some(latest_active) = serde_json::to_value(room.read_receipts())
                                .unwrap_or_default()
                                .get("latest_active")
                                .unwrap_or(&serde_json::Value::Null)
                                .get("event_id")
                            {
```